
When compiling x86/x64 toolchains the setup.py script will
automatically try to add library-dirs of the compile host.
This results in failures to compile modules on those platforms.
This first block is copied from
cross/python/patches/002-no-system-multiarch.patch

The second block allows custom -I and -L to survive into the cross environment.

*** setup.py	2018-02-04 15:40:56.000000000 -0800
--- setup.py	2018-06-24 00:01:06.695850717 -0700
***************
*** 651,663 ****
          if not CROSS_COMPILING:
              add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
              add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
          # only change this for cross builds for 3.3, issues on Mageia
          if CROSS_COMPILING:
              self.add_cross_compiling_paths()
-         self.add_multiarch_paths()
          self.add_ldflags_cppflags()
  
      def init_inc_lib_dirs(self):
          if (not CROSS_COMPILING and
                  os.path.normpath(sys.base_prefix) != '/usr' and
                  not sysconfig.get_config_var('PYTHONFRAMEWORK')):
--- 651,662 ----
***************
*** 697,709 ****
              self.lib_dirs += ['/usr/ccs/lib']
  
          # HP-UX11iv3 keeps files in lib/hpux folders.
          if HOST_PLATFORM == 'hp-ux11':
              self.lib_dirs += ['/usr/lib/hpux64', '/usr/lib/hpux32']
  
!         if MACOS:
              # This should work on any unixy platform ;-)
              # If the user has bothered specifying additional -I and -L flags
              # in OPT and LDFLAGS we might as well use them here.
              #
              # NOTE: using shlex.split would technically be more correct, but
              # also gives a bootstrap problem. Let's hope nobody uses
--- 696,708 ----
              self.lib_dirs += ['/usr/ccs/lib']
  
          # HP-UX11iv3 keeps files in lib/hpux folders.
          if HOST_PLATFORM == 'hp-ux11':
              self.lib_dirs += ['/usr/lib/hpux64', '/usr/lib/hpux32']
  
!         if MACOS or CROSS_COMPILING:
              # This should work on any unixy platform ;-)
              # If the user has bothered specifying additional -I and -L flags
              # in OPT and LDFLAGS we might as well use them here.
              #
              # NOTE: using shlex.split would technically be more correct, but
              # also gives a bootstrap problem. Let's hope nobody uses
